operator/pkg/ciliumidentity/cache.go

```diff
@@ -310,3 +308,85 @@ func (c *CIDUsageInCES) decrementUsage(cidName int64) int {
 
 	return count
 }
+
+// EnqueueTimeTracker provides a thread safe mechanism to record
+// and manage the time when items are enqueued.
+type EnqueueTimeTracker struct {
+	enqueuedAt map[string]time.Time
+	mu         lock.Mutex
+}
+
+func (e *EnqueueTimeTracker) SetEnqueueTimeIfNotSet(item string) {
+	e.mu.Lock()
+	defer e.mu.Unlock()
+	if e.enqueuedAt[item].IsZero() {
+		e.enqueuedAt[item] = time.Now()
```

Comment:
Relying on system time for the sake of synchronization is error prone in a distributed system.

The synchronization primitive in Kubernetes is etcd revision numbers via optimistic concurrency. 

My understanding is that there has been some design work on this proposal, is there a CFP that you can link to?

Reply:
Hey @asauber, thank you for your review! 

The time is only used locally to know when to schedule the cleanup. We are not actually relying in the exact time. 

CFP https://github.com/cilium/cilium/issues/27752
Draft full implementation: https://github.com/cilium/cilium/pull/33204

Reply:
I see, this seems to me to introduce some unspecified (possibly inaccurate, depending on the set system time), during which an ID continues to exist on a Node even when it should not, possibly affecting policy behavior. Is this not the case? (cc @joestringer if you have any additional thoughts)

Reply:
I don't think I could speak to the usage of this until there's code that actually uses it. It seems like the reviewers will need to look over https://github.com/cilium/cilium/pull/33383 to understand this structure before they can know that the usage of the structure doesn't bring up such concerns.